behave_reportportal/behave_agent.py

```diff
@@ -198,18 +254,121 @@ def _finish_step_scenario_based(self, step, **kwargs):
+                continue
+            msg = "Using of '{}' fixture".format(tag[len("fixture.") :])
+            if self._cfg.step_based:
+                self._step_id = self._rp.start_test_item(
+                    name=msg,
+                    start_time=timestamp(),
+                    item_type=item_type,
+                    parent_item_id=parent_item_id,
+                )
+                self._rp.finish_test_item(self._step_id, timestamp(), "PASSED")
+                continue
             self._rp.log(
-                item_id=item_id,
-                time=timestamp(),
-                level="ERROR",
-                message="Step [{keyword}]: {name} was finished with "
-                "exception:\n{exception}".format(
-                    keyword=step.keyword,
-                    name=step.name,
-                    exception=", ".join(step.exception.args),
-                ),
+                timestamp(),
+                msg,
+                level="INFO",
+                item_id=parent_item_id,
+            )
+
+    def _log_feature_cleanup(self, context):
+        layer = next(
+            iter(
+                [
+                    level
+                    for level in context._stack
+                    if level.get("@layer") == "feature"
+                ]
+            ),
+            None,
+        )
+        if not layer:
+            return
+        if not layer.get("@cleanups"):
+            return
+        for cleanup in layer.get("@cleanups"):
+            msg = "Execution of '{}' cleanup function".format(cleanup.__name__)
+            self._step_id = self._step_id = self._rp.start_test_item(
+                name=msg,
+                start_time=timestamp(),
+                item_type="AFTER_SUITE",
+                parent_item_id=self._feature_id,
             )
+            self._rp.finish_test_item(self._step_id, timestamp(), "PASSED")
+
+    def _log_scenario_cleanup(self, context):
+        layer = next(
+            iter(
+                [
+                    level
+                    for level in context._stack
+                    if level.get("@layer") == "scenario"
+                ]
+            ),
+            None,
+        )
+        if not layer:
+            return
+        if not layer.get("@cleanups"):
+            return
+        for cleanup in layer.get("@cleanups"):
+            msg = "Execution of '{}' cleanup function".format(cleanup.__name__)
+            if self._cfg.step_based:
+                self._step_id = self._step_id = self._rp.start_test_item(
+                    name=msg,
+                    start_time=timestamp(),
+                    item_type="AFTER_TEST",
+                    parent_item_id=self._scenario_id,
+                )
+                self._rp.finish_test_item(self._step_id, timestamp(), "PASSED")
+            else:
+                self._rp.log(
+                    timestamp(), msg, level="INFO", item_id=self._scenario_id
+                )
```

Comment:
Two methods contain almost identical code. That needs refactoring.

Reply:
done